### DISCUSSION TITLE
No language detected for a small repository (<40 files)

### DISCUSSION THREAD

**Discussion by ararslan** [github-linguist/linguist#6052] [Classification] [answered]
It seems to be that many of the cases discussed here where no language is shown are due to the hard limit on objects in the repository. I have a Vim plugin, https://github.com/ararslan/license-to-vim, for which no language is displayed even though the repository contains fewer than 40 total files. I've played around with linguist overrides to ensure that the .txt files are marked as vendored but that seems not to have done the trick. Apologies if I'm missing something obvious here but does anyone know what could be going on?

**Comment by lildude** [comment, marked as answer]
This is because your one and only Vimscript file is named `license.vim` which means it's matching this regex for detecting documentation:

https://github.com/github/linguist/blob/6ede82884427dceb7519d2b781fee7c51ed490a3/lib/linguist/documentation.yml#L29

You'll need to implement a `-linguist-documentation` override.

**Reply by ararslan** [reply]
Wow, that's kind of hilarious. Thanks so much!
